### PR TITLE
build: implement packaging steps for the Windows build

### PR DESCRIPTION
**Pull request by michaelsbradleyjr**
Implement a `pkg-windows` target that ultimately results in `Status.zip` being written to `pkg/`.

Note: this commit does not introduce code signing for the Windows build since that piece is still a work in progress.

`pkg-windows` creates a portable folder in `tmp/windows/dist` with the help of [`windeployqt`][windeployqt], which copies the needed portions of Qt into the folder.

Since DLL resolution is relatively inflexible, a launcher `Status.exe` is created at the top-level of the folder; the launcher opens `bin/Status.exe` while adding the portable folder's `bin/` to the `PATH`, allowing `bin/Status.exe` to resolve the DLLs in that folder.

A few additional tools need to be installed (e.g. with [scoop][scoop]) and availble in `PATH`:
* 7-zip
* dos2unix (provides unix2dos)
* findutils
* go
* rcedit
* wget

The above list builds on the tools list in PR #521, and the other requirements and instructions in that PR's description still apply.

**Why not build an installer?**

When starting work on packaging for the Windows build, my initial plan was to build an installer, and for that purpose I researched the [WiX Toolset][wix], the [Qt Installer Framework][qtif], and some other options.

I found that building an installer is a bit complex. I then recalled, from personal experience, that [Cmder][cmder]’s [Mini download][mini] is installer-less. You simply unzip the download and place the `cmder_mini` folder wherever you prefer. Such an approach was also recommended to me in one of the Nim language's community chats.

In addition to being simpler, the installer-less approach also gives installation of Status Desktop a lower profile than an installer-application would since nothing is written to the Windows registry, added to the *Add or remove programs* list, etc. I think that's a benefit given the privacy-security focus of Status, but others may feel differently so please provide feedback on this point!

[windeployqt]: https://doc.qt.io/qt-5/windows-deployment.html
[scoop]: https://scoop.sh/
[wix]: https://wixtoolset.org/
[qtif]: https://doc.qt.io/qtinstallerframework/index.html
[cmder]: https://cmder.net/
[mini]: https://github.com/cmderdev/cmder/releases/download/v1.3.15/cmder_mini.zip

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -285,18 +289,69 @@ ifdef MACOS_CODESIGN_IDENT
...
+$(STATUS_CLIENT_ZIP): nim_status_client nim_windows_launcher $(NIM_WINDOWS_PREBUILT_DLLS)
+	rm -rf pkg/*.zip
+	rm -rf tmp/windows/dist
+	mkdir -p tmp/windows/dist/Status/bin
+	mkdir -p tmp/windows/dist/Status/resources
+	mkdir -p tmp/windows/dist/Status/vendor
+	cp windows-install.txt tmp/windows/dist/Status/INSTALL.txt
+	unix2dos -k tmp/windows/dist/Status/INSTALL.txt
+	# cp LICENSE tmp/windows/dist/Status/LICENSE.txt
+	# unix2dos -k tmp/windows/dist/Status/LICENSE.txt
```
Comment (richard-ramos) [inline Makefile:316]: Are these two lines necessary?

File: Makefile [hunk 2]
```diff
@@ -285,18 +289,69 @@ ifdef MACOS_CODESIGN_IDENT
...
+
+$(NIM_WINDOWS_PREBUILT_DLLS):
+	echo -e "\e[92mFetching:\e[39m prebuilt DLLs from nim-lang.org"
+	rm -rf tmp/windows
+	mkdir -p tmp/windows/tools
+	cd tmp/windows/tools && \
+	wget https://nim-lang.org/download/dlls.zip && \
+	unzip dlls.zip
+
+nim_windows_launcher: | deps
```
Comment (richard-ramos) [inline Makefile:302]: I received this error when running this on my machine:
```
Hint:  [Link]
/usr/bin/ld: cannot find -lwinpthread
/usr/bin/ld: nimcache/debug/nim_windows_launcher/stdlib_dynlib.nim.c.o: in function `loadLib__Yq5XYz2ycX5V5B9bUM4Uyiw':
/home/richard/status/nim-status-client/vendor/nimbus-build-system/vendor/Nim/lib/pure/dynlib.nim:125: warning: Using 'dlopen' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
collect2: error: ld returned 1 exit status
Error: execution of an external program failed: 'g++   -o /home/richard/status/nim-status-client/bin/nim_windows_launcher  nimcache/debug/nim_windows_launcher/stdlib_assertions.nim.c.o nimcache/debug/nim_windows_launcher/stdlib_locks.nim.c.o nimcache/debug/nim_windows_launcher/stdlib_sharedlist.nim.c.o nimcache/debug/nim_windows_launcher/stdlib_widestrs.nim.c.o nimcache/debug/nim_windows_launcher/stdlib_system.nim.c.o nimcache/debug/nim_windows_launcher/stdlib_pathnorm.nim.c.o nimcache/debug/nim_windows_launcher/stdlib_times.nim.c.o nimcache/debug/nim_windows_launcher/stdlib_os.nim.c.o nimcache/debug/nim_windows_launcher/stdlib_dynlib.nim.c.o nimcache/debug/nim_windows_launcher/stdlib_winlean.nim.c.o nimcache/debug/nim_windows_launcher/@mnim_windows_launcher.nim.c.o  -pthread -lm -lrt  -lcrypto -lssl -Wl,-as-needed -static-libgcc -Wl,-Bstatic,--whole-archive -lwinpthread -Wl,--no-whole-archive  -ldl'
make: *** [Makefile:303: nim_windows_launcher] Error 1
```
Should i use only a windows machine for this target?

File: Makefile (hunk 1 above)
Comment (michaelsbradleyjr) [reply Makefile:316]: Once we have a LICENSE file in the root of the repo, my plan is to uncomment those lines.

File: Makefile (hunk 2 above)
Comment (michaelsbradleyjr) [reply Makefile:302]: The Windows building and packaging logic in the Makefile assumes it's being run on a Windows machine, with `mingw32-make.exe` in a Git Bash shell.

On my Windows machine, `libwinpthread-1.dll` resolves to `/c/Qt/Tools/mingw730_64/bin/libwinpthread-1.dll`.

Comment (michaelsbradleyjr) [reply Makefile:302]: > the msvc toolchain is involved in building `DOtherSide.dll`

Now I'm wondering why your build didn't fail earlier. I guess that if you tried it on Linux then DOtherSide was built the usual way there. We don't have explicit `build-macos|linux|windows` targets so maybe the `pkg-macos|linux|windows` targets should exit with an error message if the OS you're building on isn't a match for the OS the build is targeting.

Comment (richard-ramos) [reply Makefile:302]: That would be nice, because otherwise displaying these build errors could make someone think there's an issue in the build system

**Pull request merged**